examples/src/main/scala/examples/hybrid/validation/ParentBlockValidator.scala

```diff
@@ -14,16 +15,16 @@ class ParentBlockValidator(storage: HistoryStorage)
     block match {
       case powBlock: PowBlock => if (!storage.isGenesis(powBlock)) {
         //check PoW parent id ???
-        require(storage.modifierById(powBlock.parentId).isDefined, s"Parent ${Base58.encode(powBlock.parentId)} missed")
+        require(storage.modifierById(ModifierId !@@ powBlock.parentId.arr).isDefined, s"Parent ${Base58.encode(powBlock.parentId.arr)} missed")
         //check referenced PoS block exists as well
         val posBlock = storage.modifierById(powBlock.prevPosId).get
 
         //check referenced PoS block points to parent PoW block
-        require(posBlock.parentId sameElements posBlock.parentId, "ref rule broken")
+        require(posBlock.parentId.arr sameElements posBlock.parentId.arr, "ref rule broken")
```

Comment:
should be "posBlock.parentId == posBlock.parentId" , as a particular motivation to do boxing was to have "==" operator working (please check all other sameElements occurences, there are a lot of them!)